### PR TITLE
Xcode 10

### PR DESCRIPTION
**Pull request by ricardopereira**
#104

### CODE REVIEW

File: JDStatusBarNotification/JDStatusBarNotification.m [hunk 1]
```diff
@@ -289,10 +289,10 @@ - (void)dismissAnimated:(BOOL)animated;
...
     [self.overlayWindow removeFromSuperview];
     [self.overlayWindow setHidden:YES];
-    _overlayWindow.rootViewController = nil;
-    _overlayWindow = nil;
-    _progressView = nil;
-    _topBar = nil;
+    self.overlayWindow.rootViewController = nil;
+    self->_overlayWindow = nil;
+    self->_progressView = nil;
+    self->_topBar = nil;
```
Comment (ricardopereira) [inline JDStatusBarNotification/JDStatusBarNotification.m]: Fixing `JDStatusBarNotification.m:295:5: Block implicitly retains 'self';
explicitly mention 'self' to indicate this is intended behavior` warning. So I think self can be retaining until the animation finishes, right?

Comment (ricardopereira) [reply JDStatusBarNotification/JDStatusBarNotification.m]: Now that I'm looking at it more carefully, maybe a `cleanup` method would be nice.

```
- (void)cleanup {
    self.overlayWindow.rootViewController = nil;
    _overlayWindow = nil;
    _progressView = nil;
    _topBar = nil;
}
```

**Comment by calimarkus** [comment]
Thanks!

File: JDStatusBarNotification/JDStatusBarNotification.m (hunk 1 above)
Comment (calimarkus) [reply JDStatusBarNotification/JDStatusBarNotification.m]: yeah I agree, that would be even cleaner. Fell free to put it up!

**Pull request merged**